src/io.h

```diff
@@ -8,11 +8,8 @@
 #include <stdint.h>
 
 typedef struct {
```

Comment:
Add some comments to describe the interaction between RISC-V core and the memory subsystem.